lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Wallets.hs

```diff
@@ -472,6 +472,14 @@ spec = do
             let payload = payloadWith' "Secure Wallet" mnemonics24 (fromIntegral addrPoolGap)
             r <- request @ApiWallet ctx (Link.postWallet @'Shelley) Default payload
             verify r expectations
+            rl <- request @[ApiWallet] ctx (Link.listWallets @'Shelley) Default Empty
+            verify rl
+                [ expectResponseCode @IO HTTP.status200
+                , expectListSize 1
+                , expectListField 0 (#addressPoolGap . #getApiT . #getAddressPoolGap)
+                    (`shouldBe` (fromIntegral addrPoolGap))
+                ]
```

Comment:
I don't think that's what Piotr meant though. He meant to check that listing addresses returns the correct number of addresses.

Reply:
Yes, sorry for not being precise. I meant this guy -> https://input-output-hk.github.io/cardano-wallet/api/edge/#operation/listAddresses. The relationship here is that when we create wallet with address_pool_gap = N, the list address endpoint should return precisely N addresses for that wallet.

Reply:
I think should be fine now